doc/source/train/user-guides/checkpoints.rst

```diff
@@ -309,6 +316,77 @@ Lower-performing checkpoints are deleted to save storage space. By default, all
+  and an optional ``validate_config`` dictionary, which contain arguments needed for validation
+  such as the validation dataset, and returns a dictionary of metrics from the validation.
+  We recommend performing the validation with an eval-only :ref:`Trainer <train-overview-trainers>` or with
+  :func:`~ray.data.Dataset.map_batches` - see :ref:`train-validate-fn` for more details.
+* Call :func:`~ray.train.report` with your ``validate_fn`` and a ``validate_config`` dict.
+  Ray Train runs your ``validate_fn`` with the ``validation_config`` and ``checkpoint``
+  in a new Ray task. When that task completes, Ray Train associates the metrics returned by
+  ``validate_fn`` with that ``checkpoint``.
+
+The main benefits of validating with :func:`~ray.train.report` over performing the validation
+in the training loop include:
+
+* Running validation in parallel with the training loop
+* Running validation on different hardware than training
+* Leveraging :ref:`vms-autoscaling` to rent user-specified machines only for the duration of the validation
+
+.. _train-validate-fn:
+
+Writing your own validation function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here is an example ``validate_fn`` that uses a :class:`~ray.train.torch.TorchTrainer`
+to calculate average cross entropy loss on a validation set. Note that this ``report``\s
+a dummy checkpoint so that the ``TorchTrainer`` keeps the metrics. Also note that while
+the ``TorchTrainer`` is typically used for training, it can also be used for isolated
+validation, and if you are using Anyscale, all these validation-only train runs will
+show up on the `Train dashboard <https://docs.anyscale.com/monitoring/workload-debugging/train-dashboard>`.
+
+.. literalinclude:: ../doc_code/checkpoints.py
+    :language: python
+    :start-after: __validate_fn_torch_trainer_start__
+    :end-before: __validate_fn_torch_trainer_end__
+
+Here is an example ``validate_fn`` that uses :func:`~ray.data.Dataset.map_batches` to
+calculate average cross entropy loss on a validation set. To learn more about how to use
+``map_batches`` for batch inference, check out :ref:`batch_inference_home`.
+
+.. literalinclude:: ../doc_code/checkpoints.py
+    :language: python
+    :start-after: __validate_fn_map_batches_start__
+    :end-before: __validate_fn_map_batches_end__
+
+You should use ``TorchTrainer`` if:
+
+* You don't want to add a dependency on Ray Data
+* You prefer PyTorch's metrics aggregation API to Ray Data's
+
+You should use ``map_batches`` if:
+
+* You care about validation performance. Preliminary benchmarks show that ``map_batches`` is
+  faster and the Ray Data team will continue to work on OSS and proprietary performance improvements.
+* You prefer Ray Data's metrics aggregation API to PyTorch's
+
+Reporting with your validation function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once you've defined your ``validate_fn``, you can ``report`` it along with a ``validation_config``,
```

Comment:
Can you highlight the fact that you shouldn't pass the validation dataset into the original Trainer's "datasets" arg?

Reply:
Also, we should warn the user that they shouldn't capture large objects in this validate_fn, since it's run as a ray task and will serialize all captured variables.

That's why they need to package heavy things in the Checkpoint and access from shared storage later.